stream-repo-apache/src/main/java/io/getstream/client/repo/StreamActivityRepository.java

```diff
@@ -177,6 +177,18 @@ public void deleteActivityById(BaseFeed feed, String activityId) throws IOExcept
         }
     }
 
+    public void deleteActivityByForeignId(BaseFeed feed, String activityId) throws IOException, StreamClientException {
+        HttpDelete request = new HttpDelete(UriBuilder.fromEndpoint(baseEndpoint)
+                .path("feed").path(feed.getFeedSlug()).path(feed.getUserId()).path(activityId + "/")
+                .queryParam(StreamRepositoryImpl.API_KEY, apiKey)
+                .queryParam("foreign_id", Boolean.toString(true))
+                .build());
+        LOG.debug("Invoking url: '{}", request.getURI());
+        try (CloseableHttpResponse response = httpClient.execute(addAuthentication(feed, request), HttpClientContext.create())) {
```

Comment:
I was referring to a method call: https://github.com/GetStream/stream-java/blob/master/stream-repo-apache/src/main/java/io/getstream/client/repo/StreamRepositoryImpl.java#L199
It will reduce duplicate code in this class.
Btw, on okhttp implementation retries is handled automatically by the HTTP client implementation: I set it to 3.